test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/AuthorizationApplicationModelProviderTest.cs

```diff
@@ -81,6 +86,117 @@ public void CreateControllerModelAndActionModel_AllowAnonymousAttributeAddsAllow
             Assert.Single(action.Filters, f => f is AllowAnonymousFilter);
         }
 
+        [Fact]
+        public async void CreateControllerModelAndActionModel_ChangingPolicyProvider()
+        {
+            // Arrange
+            var requirements = new IAuthorizationRequirement[] {
+                new AssertionRequirement((con) => { return true; })
+            };
+            var authorizationPolicy = new AuthorizationPolicy(requirements, new string[] { "dingos" });
+            var authorizationPolicyProvider = new ChangingAuthorizationPolicyProvider();
+
+            var provider = new AuthorizationApplicationModelProvider(authorizationPolicyProvider);
+            var defaultProvider = new DefaultApplicationModelProvider(new TestOptionsManager<MvcOptions>());
+
+            var context = new ApplicationModelProviderContext(new[] { typeof(BaseController).GetTypeInfo() });
+            defaultProvider.OnProvidersExecuting(context);
+            var authorizeData = new List<IAuthorizeData> {
+                new AuthorizeAttribute("It's a Policy!")
+            };
+
+            // Act
+            provider.OnProvidersExecuting(context);
+            var firstPolicy = await AuthorizationPolicy.CombineAsync(authorizationPolicyProvider, authorizeData);
+            var secondPolicy = await AuthorizationPolicy.CombineAsync(authorizationPolicyProvider, authorizeData);
+
+            // Assert
+            Assert.Equal(1, firstPolicy.Requirements.Count);
+            Assert.Equal(2, secondPolicy.Requirements.Count);
+        }
```

Comment:
Actually never mind, since the fix affects which constructor goes into the AuthorizeFilter, the test does need to go against the DefaultApplicationModelProvider